### PR TITLE
[INFRA-2228] Add CLIENT_ID and CLIENT_SECRET env variable to pluginsite

### PR DESCRIPTION
**Pull request by olblak**
This PR inject two new environment variable to pluginsite containers in order to authenticate with Github api

### CODE REVIEW

**Comment by olblak** [comment]
Remark I didn't test this change :)

**Review by oleg-nenashev — APPROVED** [review]
Looks fine! Is it a new Client ID? Or do we share it with other services?

File: dist/profile/templates/kubernetes/resources/pluginsite/deployment.yaml.erb [hunk 1]
```diff
@@ -40,6 +40,16 @@ spec:
                       - containerPort: 8080
                         protocol: TCP
                   env:
+                      - name: CLIENT_ID
```
Comment (oleg-nenashev) [inline dist/profile/templates/kubernetes/resources/pluginsite/deployment.yaml.erb]:
```suggestion
                      - name: GITHUB_CLIENT_ID
```

File: dist/profile/templates/kubernetes/resources/pluginsite/deployment.yaml.erb [hunk 2]
```diff
@@ -40,6 +40,16 @@ spec:
...
+                        valueFrom:
+                          secretKeyRef:
+                            name: pluginsite
+                            key: client_id
+                      - name: CLIENT_SECRET
```
Comment (oleg-nenashev) [inline dist/profile/templates/kubernetes/resources/pluginsite/deployment.yaml.erb]:
```suggestion
                      - name: GITHUB_CLIENT_SECRTE
```

**Review by oleg-nenashev — COMMENTED** [review]
Would be nice to add a "GITHUB" prefix to all vars tho

**Comment by oleg-nenashev** [comment]
https://github.com/jenkins-infra/plugin-site-api/pull/59 for the production update

**Comment by olblak** [comment]
> Looks fine! Is it a new Client ID? Or do we share it with other services?

It's a new client id created at the jenkins-infra organisation level

File: dist/profile/templates/kubernetes/resources/pluginsite/secret.yaml.erb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: Secret 
+metadata: 
+    name: pluginsite
+type: Opaque
+data:
+    client_id: '<%= @parameters['client_id'] %>'
```
Comment (oleg-nenashev) [inline dist/profile/templates/kubernetes/resources/pluginsite/secret.yaml.erb]: Should it be changed here as well?

File: dist/profile/templates/kubernetes/resources/pluginsite/deployment.yaml.erb (hunk 2 above)
Comment (olblak) [reply dist/profile/templates/kubernetes/resources/pluginsite/deployment.yaml.erb]: Done

File: dist/profile/templates/kubernetes/resources/pluginsite/secret.yaml.erb (hunk 3 above)
Comment (olblak) [reply dist/profile/templates/kubernetes/resources/pluginsite/secret.yaml.erb]: Thanks for catching this

**Comment by oleg-nenashev** [comment]
I'd guess the Plugin Site API can be updated in the same PR

**Comment by olblak** [comment]
> I'd guess the Plugin Site API can be updated in the same PR

I was waiting for the new tag,

**Pull request merged**